File: LeetCode June Challenge/RunningSumof1dArray.java
public class RunningSumof1dArray{
    public int[] runningSum(int[] nums) {
        if(nums.length == 0) return null;
        for(int i = 1; i < nums.length; ++i)
            nums[i] += nums[i - 1];
        return nums;
    }
    public void displayCode(int[] nums){
        for(int i = 0; i < nums.length; ++i)
            System.out.println(nums[i]);
    }
    public static void main(String[] args) {
        int[] nums = {1,2,3,4};
        RunningSumof1dArray p = new RunningSumof1dArray();
        int[] ans = p.runningSum(nums);
        p.displayCode(ans);
        
    }
}

File: Year-2022/LeetCode June Challenge/RunningSumof1dArray.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        if(nums.empty()) return {};
        for(int i = 1; i < nums.size(); ++i){
            nums[i] += nums[i - 1];
        }
        return nums;
    }
    void displayCode(vector<int> &nums){
        for(int i = 0; i < nums.size(); ++i) cout<<nums[i]<<" ";
    }
};
int main(){
    vector<int> nums = {3,1,2,10,1};
    Solution s;
    vector<int> ans = s.runningSum(nums);
    s.displayCode(ans);
    return 0;
    
}